app/src/main/java/emu/skyline/SkylineApplication.kt

```diff
@@ -20,12 +20,26 @@ fun Context.getPublicFilesDir() : File = getExternalFilesDir(null) ?: filesDir
 class SkylineApplication : Application() {
     private external fun initializeLog(appFilesPath : String, logLevel : Int)
 
+    init {
+        instance = this
+    }
+
+    companion object {
+        private var instance: SkylineApplication? = null;
+
+        fun applicationContext() : Context {
+            return instance!!.applicationContext
+        }
```

Comment:
So I had a little bit of trouble with this since i'm used to static in java however this seems to work:

```
    init {
        instance = this
    }

    companion object {
        lateinit var instance: SkylineApplication
            private set
    }
    
     override fun onCreate() {
        super.onCreate()
        instance = this
     }
    
```

I'm not sure if this is the right way to do so though. Trying to make the context itself a getter led to problems since application context was from the actual class instead of the companion object. This is probably be just being nooby if there is any better way let me know.

Reply:
I assume this is effectively fixed given the comment below by @AbandonedCart. I did suspect this was the case but wasn't sure as I rarely deal with the application context and almost always have access to the `Activity`.